internal/default_branch.go

```diff
@@ -0,0 +1,138 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package txqualitychecks
+
+import (
+	"context"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/go-ini/ini"
+	"github.com/google/go-github/v50/github"
+	"golang.org/x/oauth2"
+)
+
+type defaultBranch struct {
+}
+
+func (d defaultBranch) Name() string {
+	return "TRG 2.01 - Default Branch"
+}
+
+func (d defaultBranch) Description() string {
+	return "Default branch must be main."
+}
+
+func (d defaultBranch) ExternalDescription() string {
+	return "https://eclipse-tractusx.github.io/docs/release/trg-2/trg-2-1"
+}
+
+func (d defaultBranch) Test() *QualityResult {
+	owner, repoName, isLocal := getRepoName()
+	repo, err := getRepoInfo(owner, repoName, isLocal)
+
+	if err != nil {
+		log.Fatalf("error %v", err)
+	}
+
+	if *repo.Fork {
+		return &QualityResult{Passed: false, ErrorDescription: "Check determined running on a fork."}
+	}
+
+	if *repo.DefaultBranch != "main" {
+		return &QualityResult{
+			Passed:           false,
+			ErrorDescription: "Default branch not set to 'main'.",
+		}
+	}
+
+	return &QualityResult{Passed: true}
+}
+
+func (d defaultBranch) IsOptional() bool {
+	return false
+}
+
+func NewDefaultBranch() QualityGuideline {
+	return &defaultBranch{}
+}
+
+// getRepoName returns repoName (string) and local (bool). repoName is either
+// read from GITHUB_REPOSITORY environment variable when running as GH Action or,
+// when executed locally, parsing local Git config.
+//
+// local it true when repoName was parsed from local git config.
+func getRepoName() (owner string, repo string, local bool) {
```

Comment:
Might be worth introducing a struct as return type. Functions with multiple return values of the same time tend to be less intuitive, since you need to check, which in which order the values are returned

Reply:
Done